R/abtestbayesian.R

```diff
@@ -410,10 +403,156 @@ ABTestBayesian <- function(jaspResults, dataset = NULL, options) {
     "p1"           = "p1",
     "p2"           = "p2"
   )
-  
+
   plotFunc <- function() {
     abtest::plot_prior(prior_par = prior_par, what = what, hypothesis = "H1")
   }
-  
+
   return(plotFunc)
 }
+
+
+.plotSequentialAnalysis.abTest <- function(abObj, thin = 1) {
+
+  # ggplot2 version of abtest::plot_sequential
+  # wherever code is copied from abtest::plot_sequential this is indicated with a start and end
+
+  # start abtest::plot_sequential
+  data <- abObj$input$data
+  ntotal <- unique(vapply(data, length, 0))
+  index <- seq(thin, ntotal, thin)
+  if (index[length(index)] != ntotal) {
+    warning("thinning hides the last observation", call. = FALSE)
+  }
+  nsteps <- length(index)
+  n <- vapply(index, function(i) data$n1[i] + data$n2[i], 0)
+
+  out <- lapply(index, FUN = abtest:::compute_ab_seq, ab = abObj, data = data, ntotal = ntotal)
+
+  col <- RColorBrewer::brewer.pal(8, name = "Dark2")[c(2, 8, 1, 6)]
+
+  xticks <- pretty(c(0, data$n1[length(data$n1)] + data$n2[length(data$n2)]))
+  xlim <- range(xticks)
+  ylim <- c(0, 1)
+  yticks <- pretty(ylim)
+  hyp_index <- abObj$input$prior_prob != 0
+
+  # end abtest::plot_sequential
+
+  names(col) <- c("H-", "H0", "H+", "H1")
+
+  hyp_index2 <- hyp_index[hyp_index]
+  nLen <- length(n)
+  dfLines <- data.frame(
+    x = rep(c(0, n), length(hyp_index2)),
+    y = numeric((1 + nLen) * length(hyp_index2)),
+    g = rep(factor(names(hyp_index2)), each = nLen + 1)
+  )
+
+  for (i in seq_along(hyp_index2)) {
+    hyp <- names(hyp_index2)[i]
+    values <- c(abObj$input$prior_prob[hyp], vapply(out, FUN = function(y) y$post_prob[hyp], FUN.VALUE = 0))
+    dfLines$y[(1 + (nLen + 1) *(i - 1)): ((nLen + 1) * i)] <- values
+  }
```

Comment:
Uh, this is tough... I wonder whether you could simplify this, but if it's too much work I guess it's fine

edit: maybe at least something like this could work?
```suggestion
  for (i in seq_along(hyp_index2)) {
    hyp <- names(hyp_index2)[i]
    values <- c(abObj$input$prior_prob[hyp], vapply(out, FUN = function(y) y$post_prob[hyp], FUN.VALUE = 0))
    dfLines$y[dfLines$g == hyp] <- values
  }
```

Reply:
changed.